ERCS/erc-7794.md

```diff
@@ -0,0 +1,820 @@
+   * @param milestoneDate The timestamp of the removed milestone.
+   */
+  event MilestoneDateRemoved(bytes32 indexed grantId, uint256 milestoneDate);
+
+  /**
+   * @dev Emitted when a disbursement is added to a milestone.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   * @param fundingToken The token used for the disbursement.
+   * @param fundingAmount The amount of the disbursement.
+   */
+  event DisbursementAdded(
+    bytes32 indexed grantId,
+    uint256 milestoneDate,
+    address indexed fundingToken,
+    uint256 fundingAmount
+  );
+
+  /**
+   * @dev Emitted when a disbursement is removed from a milestone.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   */
+  event DisbursementRemoved(bytes32 indexed grantId, uint256 milestoneDate);
+
+  /**
+   * @dev Emitted when a disbursement status is updated.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   * @param isDisbursed Boolean indicating if the disbursement has been made.
+   */
+  event DisbursementMade(
+    bytes32 indexed grantId,
+    uint256 milestoneDate,
+    bool isDisbursed
+  );
+
+  /**
+   * @dev Emitted when an external link is added to the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param link The external URL added.
+   */
+  event ExternalLinkAdded(bytes32 indexed grantId, string link);
+
+  /**
+   * @dev Emitted when an external link is removed from the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param link The external URL removed.
+   */
+  event ExternalLinkRemoved(bytes32 indexed grantId, string link);
+
+  /**
+   * @dev Stores the core details of a grant.
+   * @param id The unique identifier of the grant.
+   * @param chainid The chain ID where the grant is registered.
+   * @param community The name of the community or protocol issuing the grant.
+   */
+  struct Grant {
+    uint256 id;
+    uint256 chainid;
+    string community;
+  }
+
+  /**
+   * @dev Stores the participants related to the grant.
+   * @param grantManager The address of the manager responsible for the grant.
+   * @param grantees A set of addresses representing the grantees involved in the grant.
+   */
+  struct Participants {
+    address grantManager;
+    EnumerableSet.AddressSet grantees;
+  }
+
+  /**
+   * @dev Stores milestone-related information for the grant.
+   * @param startDate The timestamp representing the start date of the grant.
+   * @param milestonesDates A set of timestamps representing the milestone dates of the grant.
+   * @param disbursements A mapping of milestone dates to disbursements details.
+   */
+  struct Milestones {
+    uint256 startDate;
+    EnumerableSet.UintSet milestonesDates;
+    mapping(uint256 => Disbursements) disbursements;
+  }
```

Comment:
These structs don't appear in the public API of this interface (as far as I can tell from a quick look.) That makes them an implementation detail: for example an implementer could choose a different set implementation, or choose not to use sets at all. They could even choose to use struct-of-arrays representation instead of array-of-structs. So I'd suggest removing anything not part of a public event/function from the Specification section (though you should keep them in the Reference Implementation.)